File: src/main/java/ru/devobserver/domain/devices/uart/StopBits.java
package ru.devobserver.domain.devices.uart;

import com.fazecast.jSerialComm.SerialPort;

import java.util.List;

public enum StopBits {
    ONE(SerialPort.ONE_STOP_BIT),
    ONE_POINT_FIVE(SerialPort.ONE_POINT_FIVE_STOP_BITS),
    TWO(SerialPort.TWO_STOP_BITS);

    private final int code;

    StopBits(int code) {
        this.code = code;
    }

    public static StopBits fromCode(final int code) {
        return List.of(values()).stream()
                .filter(value -> value.getCode() == code)
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Can't find stop bits for code " + code));
    }

    public int getCode() {
        return code;
    }
}

File: src/main/js/domain/StopBits.js
export class StopBits {
    static ONE = { name: "1", value: 1 };
    static ONE_POINT_FIVE = { name: "1.5", value: 2 };
    static TWO = { name: "2", value: 3 };

    static values() {
        return [this.ONE, this.ONE_POINT_FIVE, this.TWO];
    }
}